### PR TITLE
Feature/update packages

### PR DESCRIPTION
**Pull request by Dassderdie**
Updates all packages.

* Fix all vulnerabilities (in favor of #576)
* Adds support for `npm@9`
    *   `install-links` is now, by default, `false` instead of `true`. This means that the initial build works fine, but updates to `shared` are not propagated to the `frontend` and `backend` (via symbolic links).
* Jest configs were updated to remove deprecation warnings
* Prettier had some formatting changes with pipes
* OpenLayers "fixed a bug":
    * > `ol/style/Image` and subclasses displacement is no longer scaled with the image. If you previously expected this unintended behavior, you should now increase the displacement when setting the scale. (https://github.com/openlayers/openlayers/releases/tag/v7.0.0)
    * We relied on this. Therefore I had to write extra code to get the patientStatusCircles to stay in the patient, independent from the `zoom`
* We now have Angular 15. Nothing too exciting for us there, I think. But better stack traces and some improvements to the Language server are always nice to have.
* uuid.v4 got 4.3 times faster
* Multiple packages dropped IE, node 10, Safari 10, etc.

[TypeScript 4.9 will probably be supported with Angular 15.1](https://github.com/angular/angular/issues/48097#issuecomment-1318168405)

### CODE REVIEW

File: frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/style-helper/circle-style-helper.ts [hunk 1]
```diff
@@ -14,8 +14,10 @@ export class CircleStyleHelper extends StyleHelper<Style, Feature> {
     constructor(
         private readonly getProperties: (
             feature: Feature
-        ) => Exclude<Options, { scale: any }> | undefined,
-        private readonly scale: number
+        ) => Exclude<Options, { scale: any; displacement: any }> | undefined,
+        // THese two properties are zoom-dependent
```
Comment (ClFeSc) [inline frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/style-helper/circle-style-helper.ts:18]:
```suggestion
        // These two properties are zoom-dependent
```

File: frontend/src/styles.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@import '~bootstrap/scss/bootstrap.scss';
+@import '../node_modules/bootstrap/scss/bootstrap.scss';
```
Comment (ClFeSc) [inline frontend/src/styles.scss:1]: Is this the best approach? It looks a bit strange.

File: frontend/.browserslistrc [hunk 3]
```diff
@@ -1,16 +0,0 @@
-# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
```
[the remark this answers is not in the document]
Comment (Dassderdie) [reply frontend/.browserslistrc:1]: This was done by the automatic migration to angular 15.

The reason was that it is the default Browserlist.

Should we add it in again to have the supported browser versions somewhere?

File: frontend/src/styles.scss (hunk 2 above)
Comment (Dassderdie) [reply frontend/src/styles.scss:1]: The `~` is deprecated. I don't like it either, but I don't know any better alternatives...

File: frontend/.browserslistrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (ClFeSc) [reply frontend/.browserslistrc:1]: If it was automatically removed I think we can leave it out.

File: backend/package.json [hunk 4]
```diff
@@ -34,36 +34,36 @@
...
+        "dotenv": "^16.0.3",
         "envalid": "^7.3.1",
-        "express": "^4.18.1",
+        "express": "^4.18.2",
         "lodash-es": "4.17.21",
-        "pg": "^8.7.3",
-        "socket.io": "^4.5.1",
-        "supertest": "^6.2.4",
-        "typeorm": "^0.3.7"
+        "pg": "^8.8.0",
```
[the remark this answers is not in the document]
Comment (Dassderdie) [reply backend/package.json:44]: I tried it out but received an error because it was not installed.

**Review by ClFeSc — APPROVED** [review]
Tested it and found nothing out of the ordinary.

**Pull request merged**